src/ai/ai.cpp

```diff
@@ -4440,14 +4453,17 @@ void assign_targets(sys::state& state, dcon::nation_id n) {
 
 	// organize attack stacks
 	bool is_at_war = state.world.nation_get_is_at_war(n);
-	int32_t max_attacks_to_make = is_at_war ? (ready_count + 3) / 4 : ready_count; // not at war -- allow all stacks to attack rebels
+	int32_t max_attacks_to_make = is_at_war ? (ready_count + 1) / 3 : ready_count; // not at war -- allow all stacks to attack rebels
 	auto const psize = potential_targets.size();
 
+	for(uint32_t i = 0; i < psize; ++i) {
+		potential_targets[i].strength_estimate = estimate_enemy_defensive_force(state, potential_targets[i].location, n) + 0.00001f;
```

Comment:
why are you doing it all up here? The reason that it was delayed to below was that it is possible that not all of them will actually be looked at, so it is more efficient not to compute the strength unless we actually need that number